lib/python/picongpu/picmi/diagnostics/openpmd.py

```diff
@@ -0,0 +1,154 @@
+"""
+This file is part of PIConGPU.
+Copyright 2021-2025 PIConGPU contributors
+Authors: Masoud Afshari
+License: GPLv3+
+"""
+
+from ...pypicongpu.output.openpmd import OpenPMD as PyPIConGPUOpenPMD
+from .timestepspec import TimeStepSpec
+from .openpmd_sources.source import Source
+
+import typeguard
+from typing import Optional, Dict, Union
+
+
+@typeguard.typechecked
+class OpenPMD:
+    """
+    Specifies the parameters for the openPMD plugin in PIConGPU simulations.
+
+    This plugin outputs simulation data (fields and particles) to disk using the openPMD API,
+    with configurable periods, data sources, and backend settings.
+
+    Attention: **period is mandatory.**
+
+    Parameters
+    ----------
+    period: TimeStepSpec
+        Specify on which time steps to output data.
+        Unit: steps (simulation time steps). Required.
+
+    source: Source, optional
+        Data sources and filters to dump (e.g., Source([ChargeDensity(filter="filterX"), "species_all"])).
```

Comment:
I am not sure why we need that accumulation object here, from my point of view we should pass a list of objects directly here.